### ISSUE TITLE
IndentationConsistency with "EnforcedStyle: rails" does not work in Rails concerns

### ISSUE THREAD

**Issue by 5t111111** [bbatsov/rubocop#3494]
In Rails concerns, IndentationConsistency with "EnforcedStyle: rails" does not detect incorrect indentation.

--------

## Expected behavior

```ruby
concern :Authenticatable do
  def foo
    puts "foo"
  end

  private

  def bar
    puts "bar"
  end
end
```

In the above code, the method `bar` should be indented to 1 more depth, but Rubocop does not complain about this.

## Actual behavior

Inconsistent indentation should be detected.

## Steps to reproduce the problem

1. Set `IndentationConsistency` with `EnforcedStyle: rails` in your Rubocop config
2. Write the code contains inconsistent indentation in a concern block (the one in "Expected behavior" for example")
3. Check the code by running Rubocop

## RuboCop version

Include the output of `rubocop -V`:

```
$ rubocop -V
0.42.0 (using Parser 2.3.1.2, running on ruby 2.3.1 x86_64-darwin15)
```
